### ISSUE TITLE
Check box as to whether a global variable is shown isn't maintained across sprites

### ISSUE THREAD

**Issue by ToonTalk** [jmoenig/Snap--Build-Your-Own-Blocks#2141]
I obtained
![image](https://user-images.githubusercontent.com/2138696/42328116-6af56d30-8065-11e8-8ba6-4302ba9cb421.png)
by creating x, creating a new sprite, selecting it, toggling show x, back to first sprite, toggle again. No check mark and yet it is shown (and vice versa).

**Comment by jguille2** [comment]
Yes, I can reproduce this issue.

We need to add a `ide.flushBlocksCache(cat)'  to the _SpriteMorph.prototype.toggleWatcher_ function.
This will restart palettes with their defaults (removing changes... for example input block changes) but I think it is the best way.

@jmoenig. If this is right, I'll submit a PR to implement it faster.

**Comment by jmoenig** [comment]
yes, thank you both for noticing this. Joan, your solution would work, but it would also slow things down. I need to re-design the whole checkbox mechanism for variable watchers anyway (I want to remove the mechanism that prevents you from creating same-named eclipsed variables in different scopes (global vs. sprite-local). So, let's just leave this open until that other work is done (which is on my list, but not super-duper short-listed.

**Comment by jguille2** [comment]
Ok Jens,

Only a suggestion (if you want to re-design this).

Maybe we can prevent sprites to store their own full cached palette. We can construct (re-fresh) palettes using a _generic.paletteCache[category]_ and a _sprite.paletteCache[category]_. Into the _sprite.paletteCache_ we would have only local elements (if any) and into _generic_ one, all the common elements.
This would be a big change, and I know there are more important elements to keep in mind (stage vs sprites, clones, inherited properties...) and other details (for example scrollbar position is now stored for each sprite...). But I think is a good idea to separate a generic cache (for all) and a local cache (for each).
